Mage.Tests/src/test/java/org/mage/test/cards/single/woe/AshiokWickedManipulatorTest.java

```diff
@@ -0,0 +1,313 @@
+
+    /**
+     * Ashiok, Wicked Manipulator
+     * {3}{B}{B}
+     * Legendary Planeswalker — Ashiok
+     * <p>
+     * If you would pay life while your library has at least that many cards in it, exile that many cards from the top of your library instead.
+     * +1: Look at the top two cards of your library. Exile one of them and put the other into your hand.
+     * −2: Create two 1/1 black Nightmare creature tokens with "At the beginning of combat on your turn, if a card was put into exile this turn, put a +1/+1 counter on this creature."
+     * −7: Target player exiles the top X cards of their library, where X is the total mana value of cards you own in exile.
+     * <p>
+     * Loyalty: 5
+     */
+    private static final String ashiok = "Ashiok, Wicked Manipulator";
+
+    /**
+     * Final Payment
+     * {W}{B}
+     * Instant
+     * <p>
+     * As an additional cost to cast this spell, pay 5 life or sacrifice a creature or enchantment.
+     * <p>
+     * Destroy target creature.
+     */
+    private static final String finalPayment = "Final Payment";
+
+    /**
+     * Well sometimes you do need a 2/2 vanilla with mana value 2.
+     */
+    private static final String lion = "Silvercoat Lion";
+
+    /**
+     * Bolas's Citadel
+     * {3}{B}{B}{B}
+     * Legendary Artifact
+     * <p>
+     * You may look at the top card of your library any time.
+     * <p>
+     * You may play lands and cast spells from the top of your library. If you cast a spell this way, pay life equal to its mana value rather than pay its mana cost.
+     * <p>
+     * {T}, Sacrifice ten nonland permanents: Each opponent loses 10 life.
+     */
+    private static final String citadel = "Bolas's Citadel";
+
+    /**
+     * Lurking Evil
+     * {B}{B}{B}
+     * Enchantment
+     * <p>
+     * Pay half your life, rounded up: Lurking Evil becomes a 4/4 Phyrexian Horror creature with flying.
+     */
+    private static final String lurking = "Lurking Evil";
+
+    /**
+     * Arrogant Poet
+     * {1}{B}
+     * Creature — Human Warlock
+     * <p>
+     * Whenever Arrogant Poet attacks, you may pay 2 life. If you do, it gains flying until end of turn.
+     */
+    private static final String poet = "Arrogant Poet";
+
+    // Move all cards from the player's library into its graveyard (exile is not where we want them here.).
+    private void millAllLibrary(Player player) {
+        Set<Card> cards = currentGame.getPlayer(player.getId()).getLibrary().getTopCards(currentGame, 100);
+        player.moveCards(cards, Zone.GRAVEYARD, null, currentGame);
+    }
+
+    @Test
+    public void emptyALibrary() {
+        setStrictChooseMode(true);
+
+        skipInitShuffling();
+        millAllLibrary(playerA);
+        addCard(Zone.LIBRARY, playerA, lion, 10);
+
+        setStopAt(1, PhaseStep.UPKEEP);
+        execute();
+
+        assertLibraryCount(playerA, 10);
+        assertLibraryCount(playerA, lion, 10);
```

Comment:
maybe assertExileCount 0 as well? Also I think it's redundant to count 10 cards if you're counting 10 lions

Reply:
Actually it is helpful to test both, there is only 10 cards in the library and the initialization with only lions did work properly (only lions are there)